src/context/toast.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { message } from "antd";
+import { MessageType } from "antd/es/message";
+
+type Message = (toastMessage: string) => MessageType;
+
+interface ToastDispatchValue {
+  success: Message;
+  warning: Message;
+  error: Message;
+  info: Message;
+}
+
+const MessagesDispatchContext = React.createContext<ToastDispatchValue | null>(
+  null
+);
+
+enum Toast {
+  success = "success",
+  warning = "warning",
+  error = "error",
+  info = "info"
+}
+
+const showToastMessage = (t: keyof typeof Toast) => (toastMessage: string) =>
```

Comment:
this can just be t: Toast

Reply:
<img width="602" alt="Screen Shot 2020-02-11 at 7 05 20 PM" src="https://user-images.githubusercontent.com/15262143/74290988-7b782980-4d01-11ea-90b4-b1390f5b920e.png">

that is a natural thought to have but unfortunately TS does not work that way. It has to be `keyof typof enum`